metadata/content.go

```diff
@@ -383,14 +398,20 @@ func (cs *contentStore) Writer(ctx context.Context, opts ...content.WriterOpt) (
 				return nil
 			}
 
-			if st, err := cs.Store.Info(ctx, wOpts.Desc.Digest); err == nil {
-				// Ensure the expected size is the same, it is likely
-				// an error if the size is mismatched but the caller
-				// must resolve this on commit
-				if wOpts.Desc.Size == 0 || wOpts.Desc.Size == st.Size {
-					shared = true
-					wOpts.Desc.Size = st.Size
+			switch cs.policy {
+			case PolicyShared:
```

Comment:
should we consider the isolated case in one namespace?

https://github.com/containerd/containerd/blob/e32cb43e363065ae29e989b0c74c500a8d501a75/metadata/content.go#L388-L399

Reply:
Sounds like `alwayspull`, it seems useful in some special scenario. However I don't think it's appropriate using it as a global configuration. And a finer grained configuration would lead to configuration complexity, doesn't it ?

Reply:
sure. it make senses.